lib/cli/commands/generate.js

```diff
@@ -31,21 +34,142 @@ addons).
   }
 
   generateBlueprint(blueprintName, argTokens) {
-    const Blueprint = require(`../blueprints/${ blueprintName }`).default;
-    let blueprint = new Blueprint();
-    let blueprintArgs = this.parseArgs.call(blueprint, argTokens.slice(1));
-    blueprint.generate(blueprintArgs);
+    const result = this.findBlueprint(blueprintName);
+
+    if (result) {
+      let blueprint = result.instance;
+      let blueprintArgs = this.parseArgs.call(blueprint, argTokens.slice(1));
+      blueprint.generate(blueprintArgs);
+    } else {
+      ui.info(`Blueprint '${ blueprintName }' doesn't exist.`);
+    }
+  }
+
+  findBlueprint(name) {
+    const split = name.split(':');
+    const hasNamespace = split.length > 1;
+    const namespace = hasNamespace ? split[0] : undefined;
+    const blueprintName = hasNamespace ? split.slice(1).join(':') : split[0];
+    const addonData = discoverAddons(process.cwd());
+    const addons = topsort(addonData).map((data) => {
+      return new Addon(data.name, data.dir);
+    });
+    let blueprints = addons.reduce((all, addon) => {
+      if (addon.hasBlueprint(blueprintName)) {
+        let blueprint = addon.getBlueprint(blueprintName);
+        if (blueprint) {
+          all.push(blueprint);
+        }
+      }
+
+      return all;
+    }, []);
+
+    // Add local cli blueprint if it exists
+    try {
+      let basePath = `../blueprints/${ blueprintName }`;
+      let localBlueprint = require(basePath).default;
+
+      blueprints.unshift({
+        namespace: 'denali',
+        path: basePath,
+        instance: localBlueprint
+      });
+    } catch (e) {
+      // noop
+    }
+
+    return hasNamespace ? find(blueprints, 'namespace', namespace) : blueprints.slice(-1)[0];
   }
 
   printHelp() {
     super.printHelp();
-    ui.info('Available Blueprints:');
+    const addons = discoverAddons(process.cwd());
+
+    if (addons.length) {
```

Comment:
Mind changing this to an explicit `addons.length > 0` check? I'd like to avoid conditionals that rely on type coercions, like 0 being falsey here.